### PR TITLE
Regularsequence enhancement

### PR DESCRIPTION
**Pull request by AlexSananka**
@africanmathsinitiative/developers this is ready for review.
This are minor changes.I have set the defaults correctly for the `ucrInputs` and some changes on the text in the preview box suggested by @rdstern in #3416

### CODE REVIEW

File: instat/dlgRegularSequence.vb [hunk 1]
```diff
@@ -202,9 +200,11 @@ Public Class dlgRegularSequence
                 vecSequence = frmMain.clsRLink.RunInternalScriptGetValue(strRCommand, bSilent:=True).AsNumeric
                 ucrBase.clsRsyntax.SetAssignTo(strAssignToName:=ucrNewColumnName.GetText, strTempDataframe:=ucrSelectDataFrameRegularSequence.cboAvailableDataFrames.Text, strTempColumn:=ucrNewColumnName.GetText)
                 If iLength < ucrSelectDataFrameRegularSequence.iDataFrameLength Then
-                    txtMessage.Text = "Sequence has been extended by repeating to match the length of the data frame."
+                    txtMessage.Text = "Sequence extended to match
+the length of the data frame."
                 ElseIf iLength > ucrSelectDataFrameRegularSequence.iDataFrameLength Then
-                    txtMessage.Text = "Sequence has been truncated to match the length of the data frame."
+                    txtMessage.Text = "Sequence truncated to match
+the length of the data frame."
```
Comment (dannyparsons) [inline instat/dlgRegularSequence.vb]: Does this work to get it to display on two lines?

File: instat/dlgRegularSequence.vb [hunk 2]
```diff
@@ -108,20 +95,31 @@ Public Class dlgRegularSequence
...
-        clsSeqFunction.AddParameter("from", 1)
+        ' clsSeqFunction.AddParameter("from", 1)
         clsSeqFunction.AddParameter("to", ucrSelectDataFrameRegularSequence.iDataFrameLength)
-        clsSeqFunction.AddParameter("by", 1)
+        '  clsSeqFunction.AddParameter("by", 1)
 
         clsRepFunction.SetRCommand("rep")
         clsRepFunction.AddParameter("x", clsRFunctionParameter:=clsSeqFunction)
-        clsRepFunction.AddParameter("each", 1)
+        ' clsRepFunction.AddParameter("each", 1)
```
Comment (dannyparsons) [inline instat/dlgRegularSequence.vb]: why have these been commented out?

File: instat/dlgRegularSequence.vb (hunk 1 above)
Comment (AlexSananka) [inline instat/dlgRegularSequence.vb]: yeah this works fine.

File: instat/dlgRegularSequence.vb (hunk 2 above)
Comment (AlexSananka) [inline instat/dlgRegularSequence.vb]: Because I have set their default values elsewhere.

Comment (AlexSananka) [inline instat/dlgRegularSequence.vb]: I think I need to delete them.

File: instat/dlgRegularSequence.vb (hunk 1 above)
Comment (dannyparsons) [inline instat/dlgRegularSequence.vb]: actually can you change this to use `& Environment.NewLine &` instead? I think that is a more solid way to do this as different machines do lines slightly different if you just type it out.

File: instat/dlgRegularSequence.vb (hunk 2 above)
Comment (dannyparsons) [inline instat/dlgRegularSequence.vb]: Sorry I don't see where they are being done? This should be the correct place to set defaults.

Comment (AlexSananka) [inline instat/dlgRegularSequence.vb]: I have done this by setting` objNewDefaultState` of the ucrInput.

File: instat/dlgRegularSequence.vb [hunk 3]
```diff
@@ -48,7 +48,7 @@ Public Class dlgRegularSequence
 
         ucrInputInStepsOf.SetParameter(New RParameter("by", 2))
         ucrInputInStepsOf.AddQuotesIfUnrecognised = False
-        ucrInputInStepsOf.SetValidationTypeAsNumeric(dcmMin:=0)
+        ucrInputInStepsOf.SetValidationTypeAsNumeric(dcmMin:=1)
```
Comment (dannyparsons) [inline instat/dlgRegularSequence.vb]: This isn't the minimum It can be any value greater than 0

File: instat/dlgRegularSequence.vb [hunk 4]
```diff
@@ -202,9 +202,9 @@ Public Class dlgRegularSequence
                 vecSequence = frmMain.clsRLink.RunInternalScriptGetValue(strRCommand, bSilent:=True).AsNumeric
                 ucrBase.clsRsyntax.SetAssignTo(strAssignToName:=ucrNewColumnName.GetText, strTempDataframe:=ucrSelectDataFrameRegularSequence.cboAvailableDataFrames.Text, strTempColumn:=ucrNewColumnName.GetText)
                 If iLength < ucrSelectDataFrameRegularSequence.iDataFrameLength Then
-                    txtMessage.Text = "Sequence has been extended by repeating to match the length of the data frame."
+                    txtMessage.Text = "Sequence extended to match the length of the data frame."
                 ElseIf iLength > ucrSelectDataFrameRegularSequence.iDataFrameLength Then
-                    txtMessage.Text = "Sequence has been truncated to match the length of the data frame."
+                    txtMessage.Text = "Sequence truncated to match the length of the data frame."
```
Comment (dannyparsons) [inline instat/dlgRegularSequence.vb]: where are the `Environment.NewLine` I suggested?

File: instat/dlgRegularSequence.vb [hunk 5]
```diff
@@ -122,6 +109,19 @@ Public Class dlgRegularSequence
...
+    Private Sub SetRCodeForControls(bReset As Boolean)
+        bUpdateBy = False
+        ucrNewColumnName.SetRCode(ucrBase.clsRsyntax.clsBaseFunction, bReset)
+        ucrPnlSequenceType.SetRCode(ucrBase.clsRsyntax.clsBaseFunction, bReset)
+        ucrInputFrom.SetRCode(clsSeqFunction, bReset)
+        ucrInputTo.SetRCode(clsSeqFunction, bReset)
+        ucrNudRepeatValues.SetRCode(clsRepFunction, bReset)
+        If bReset AndAlso (ucrInputFrom.GetText < ucrInputTo.GetText) Then
+            ucrInputInStepsOf.SetRCode(clsSeqFunction, bReset)
+        End If
```
Comment (dannyparsons) [inline instat/dlgRegularSequence.vb]: If we need to treat the in steps of slightly differently then we could not set the R code at all and set it manually somewhere else. Will think about this a bit more.

File: instat/dlgRegularSequence.vb (hunk 3 above)
Comment (AlexSananka) [inline instat/dlgRegularSequence.vb]: did you mean any value greater than or less than 0?
And can it be 0?

File: instat/dlgRegularSequence.vb (hunk 4 above)
Comment (AlexSananka) [inline instat/dlgRegularSequence.vb]: I am adding this.

File: instat/dlgRegularSequence.vb (hunk 3 above)
Comment (dannyparsons) [inline instat/dlgRegularSequence.vb]: no it can't be 0. You should set bIncludeMin to False so that 0 is not allowed.

**Comment by rdstern** [comment]
If it were 0 would that be a simple (alternative?) way of filling a column with a single value?

**Comment by dannyparsons** [comment]
Yes but I think that would be more work for little gain at this stage.

**Comment by AlexSananka** [comment]
@stevenndungu please test this I believe it's ready for today's release.

**Comment by stevenndungu** [comment]
@dannyparsons  this is ready for you to review

**Pull request merged**